federation/pkg/federation-controller/service/servicecontroller.go

```diff
@@ -179,10 +176,6 @@ func New(federationClient fedclientset.Interface) *ServiceController {
 			svc := obj.(*v1.Service)
 			orphanDependents := false
 			err := client.Core().Services(svc.Namespace).Delete(svc.Name, &metav1.DeleteOptions{OrphanDependents: &orphanDependents})
-			// IsNotFound error is fine since that means the object is deleted already.
```

Comment:
It is already taken care of here. https://github.com/kubernetes/kubernetes/blob/master/federation/pkg/federation-controller/util/federated_updater.go#L125